### PR TITLE
First steps in abstracting out ClusterResources

### PR DESCRIPTION
**Pull request by kris-nova**
Interface out ClusterResources for

 - Toolbox dump
 - Delete

Addresses https://github.com/kubernetes/kops/issues/1077
More work for https://github.com/kubernetes/kops/issues/1079

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kops/1594)

### CODE REVIEW

**Comment by kris-nova** [comment]
@k8s-bot test this

File: upup/pkg/kutil/cluster_resources.go [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+func (c *AwsCluster) ListResources() (map[string]*ResourceTracker, error) {
+	cloud := c.Cloud.(awsup.AWSCloud)
+
+	resources := make(map[string]*ResourceTracker)
+
+	// These are the functions that are used for looking up
+	// cluster resources by their tags.
+	listFunctions := []listFn{
+		// EC2
+		ListInstances,
```
Comment (chrislovecnm) [inline upup/pkg/kutil/cluster_resources.go]: How about asg, the thing u make an asg will look up, nag gw, elastic ip, route table. I am sure you have them just not matching the names to the API. EBS volumes as well?

File: upup/pkg/kutil/cluster_resources.go [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+		ListKeypairs,
+		ListSecurityGroups,
```
Comment (chrislovecnm) [inline upup/pkg/kutil/cluster_resources.go]: Is this a code move or new code?

File: upup/pkg/kutil/cluster_resources.go [hunk 3]
```diff
@@ -0,0 +1,170 @@
...
+		ListVolumes,
+		// EC2 VPC
```
Comment (chrislovecnm) [inline upup/pkg/kutil/cluster_resources.go]: Ahhh you moved it from delete

**Comment by chrislovecnm** [comment]
Also CI is not happy

File: upup/pkg/kutil/cluster_resources.go (hunk 1 above)
Comment (kris-nova) [inline upup/pkg/kutil/cluster_resources.go]: Trying not to change delete logic in this PR - just change the way we interface with it..

File: upup/pkg/kutil/cluster_resources.go (hunk 2 above)
Comment (kris-nova) [inline upup/pkg/kutil/cluster_resources.go]: Code move

File: upup/pkg/kutil/cluster_resources.go (hunk 3 above)
Comment (kris-nova) [inline upup/pkg/kutil/cluster_resources.go]: Indeed :)

**Comment by kris-nova** [comment]
@k8s-bot test this

**Comment by kris-nova** [comment]
@k8s-bot test this

**Pull request merged**